src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/NameAnalyzer.cs

```diff
@@ -51,5 +51,17 @@ private static string GetClosestString(string name, IEnumerable<string> availabl
                 }
             }
         }
+
+        private static string GetCleanedFunctionName(string functionName)
```

Comment:
I'd like to see this in FunctionAnalyzer, so we can store the function name from CallActivity* correctly. This will allow our ArgumentAnalyzer and FunctionReturnTypeAnalyzers to work as expected without adding these extra method calls. 

Line 73 in FunctionAnalyzer stores that function name. Currently it trims the parenthesis off of the string. You could modify this method to either 1. trim nameof() off, like you currently do or 2. trim the parenthesis if the string doesn't start with nameof(.

Reply:
`GetCleanedFunctionName` is now an extension method which is called in the FunctionAnalyzer